apps/emqx_rule_engine/src/emqx_rule_funcs.erl

```diff
@@ -252,6 +252,9 @@
     timezone_to_offset_seconds/1
 ]).
 
+%% System functions
+-export([getenv/1]).
```

Comment:
Environment variables are often used to store secrets, I would be very cautious to have a function that exposes them through the rule engine. Especially if its input may somehow by templated from the contents on MQTT message.

Reply:
Yes, I agree with you, but this is a customer requirement. They use environment variables to deploy EMQX, and they want to be able to pull the kafka topic from the environment variable.
In the first [plan](https://github.com/emqx/emqx/pull/13499), we created a new app that allowed customization of predefined variables in placeholders. However, after review, it was considered too complex.

Reply:
Perhaps this function should be disabled by default, with possibility to enable it via feature flag? 
Otherwise one customer's demands can create vulnerability of all EMQX users...

Reply:
sorry, missed this point.
will add a forced prefix instead.

Reply:
https://github.com/emqx/emqx/pull/13540